cmd/createorupdate/update.go

```diff
@@ -23,6 +36,19 @@ func update(ctx context.Context, cs *api.OpenShiftManagedCluster, p api.Plugin)
 	vmc := compute.NewVirtualMachineScaleSetVMsClient(cs.Properties.AzProfile.SubscriptionID)
 	vmc.Authorizer = authorizer
 
+	for role, scale := range scaleOps {
```

Comment:
Post-deploy, detect new VMs and wait for them to be ready.  The kernel for all of this is in updatePlusOne, it just needs to be factored out.  I think it's much better for us just to await readiness on new VMs.

Reply:
*Then*, if we were just doing a scale, return early.

Reply:
Does it matter if we just check readiness for all VMs now? Tidying up to check new VMs only can be a follow-up?

Reply:
Fixed the readiness check. Still outstanding is whether we scale+upgrade in one go.